diff --git a/include/extractor/class_data.hpp b/include/extractor/class_data.hpp
index 99cb02e03..9eeb9877a 100644
--- a/include/extractor/class_data.hpp
+++ b/include/extractor/class_data.hpp
@@ -10,10 +10,10 @@
 namespace osrm::extractor
 {
 
-using ClassData = std::uint8_t;
+using ClassData = std::uint16_t;
 constexpr ClassData INAVLID_CLASS_DATA = std::numeric_limits<ClassData>::max();
-static const std::uint8_t MAX_CLASS_INDEX = 8 - 1;
-static const std::uint8_t MAX_EXCLUDABLE_CLASSES = 8;
+static const std::uint8_t MAX_CLASS_INDEX = 16 - 1;
+static const std::uint8_t MAX_EXCLUDABLE_CLASSES = 16;
 
 inline bool isSubset(const ClassData lhs, const ClassData rhs) { return (lhs & rhs) == lhs; }
 
